relay-server/src/metrics.rs

```diff
@@ -0,0 +1,200 @@
+            RelayHistograms::ProjectStatePending => "project_state.pending",
+            RelayHistograms::ProjectStateRequestBatchSize => "project_state.request.batch_size",
+            RelayHistograms::ProjectStateReceived => "project_state.received",
+            RelayHistograms::ProjectStateCacheSize => "project_cache.size",
+        }
+    }
+}
+
+/// Timer metrics used by Relay
+pub enum RelayTimers {
+    /// The time spent deserializing an event from a JSON byte array into a `relay_general::protocol::Event`.
+    EventProcessingDeserialize,
+    /// Time spent running event processors on an event.
+    /// Event processing happens before filtering.
+    EventProcessingProcess,
+    /// Time spent running filtering on an event.
+    EventProcessingFiltering,
+    /// Time spent checking for rate limits in Redis.
+    /// Note that not all events are checked against Redis. After an event is rate limited
+    /// for period A, any event using the same key coming during period A will be automatically
+    /// rate limited without checking against Redis.
+    EventProcessingRateLimiting,
+    /// Time spent in PII for the current event.
+    EventProcessingPii,
+    /// Time spent converting the event from an Annotated<Event> into a String containing the JSON
+    /// representation of the event.
+    EventProcessingSerialization,
+    /// Represents the time spent between receiving the event in Relay (i.e. beginning of the
+    /// request handling) up to the time before starting synchronous processing in the EventProcessor.
+    /// This is effectively the time between the event arriving in Relay and the event being ready
+    /// to be processed.
+    EventWaitTime,
+    /// This is the time the event spends in the EventProcessor (i.e. the sync processing of the
+    /// event).
+    /// It includes decoding the event envelope and the times spent in EventProcessingProcess,
+    /// `EventProcessingFiltering`, `EventProcessingRateLimiting`, `EventProcessingPii` and
+    /// EventProcessingSerialization.
+    EventProcessingTime,
+    /// The total time an event spends in Relay from the time it is received until it finishes
+    /// processing.
+    EventTotalTime,
+    /// The total time spent during `ProjectCache.fetch_states` in which eviction of outdated
+    /// projects happens.
+    ProjectStateEvictionDuration,
+    /// The total time spent during `ProjectCache.fetch_states` spent waiting for all ProjectState
+    /// requests to resolve. During a fetch_states request, we pick up to max_num_requests *
+    /// max_num_project_states_per_request projects that need their state updated and batch
+    /// them into max_num_requests requests. This metric represents the time spent from issuing
+    /// the first request until all requests are finished.
+    ProjectStateRequestDuration,
+    /// The total time spent getting the project id from upstream.
+    /// The project id is only fetched once and it is not refreshed.
```

Comment:
Similar to `ProjectIdRequest`, let's mention that this is about the legacy store endpoint.